Tasks/JavaToolInstaller/task.json

```diff
@@ -0,0 +1,155 @@
+    ],
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 0,
+        "Minor": 1,
+        "Patch": 0
+    },
+    "preview": true,
+    "satisfies": ["Java"],
+    "demands": [],
+    "groups": [
+        {
+            "name":"JavaInAzureGroup",
+            "displayName": "Download Java from Azure",
+            "isExpanded": true
+        }
+    ],
+    "instanceNameFormat": "Use Java $(versionSpec)",
+    "inputs": [
+        {
+            "name": "versionSpec",
+            "type": "string",
+            "label": "JDK Version",
+            "defaultValue": "1.8.0",
+            "required": true,
+            "helpMarkDown": "A number that specifies the JDK version to make available on the path."
+        },
+        {
+            "name": "jdkSource",
+            "type": "pickList",
+            "label": "JDK source",
+            "required": true,
+            "options": {
+                "AzureStorage": "Azure Storage",
+                "LocalDirectory": "Local Directory"
+            },
+            "helpMarkDown": "Source for the compressed JDK."
+        },
+        {
+            "name": "jdkPath",
+            "type": "filepath",
+            "label": "JDK path",
+            "required": true,
+            "visibleRule": "jdkSource == LocalDirectory",
+            "helpMarkDown": "Path to where the compressed JDK is located. The path could be in your source repository or a local path on the agent."
+        },
+        {
+            "name": "ConnectedServiceNameARM",
+            "type": "connectedService:AzureRM",
+            "label": "Azure Subscription",
+            "required": true,
+            "visibleRule": "jdkSource == AzureStorage",
+            "helpMarkDown": "Choose the Azure Resource Manager subscription for the JDK."
+        },
+        {
+            "name": "storageAccountName",
+            "type": "pickList",
+            "label": "Storage Account Name",
+            "required": true,
+            "properties": {
+                "EditableOptions":"True"
+            },
+            "visibleRule": "jdkSource == AzureStorage",
+            "helpMarkDown": "Azure Classic and Resource Manager storage accounts are listed. Select the Storage account name in which the JDK is located."
+        },
+        {
+            "name": "containerName",
+            "type": "string",
+            "label": "Container Name",
+            "required": true,
+            "visibleRule": "jdkSource == AzureStorage",
+            "helpMarkDown": "Name of the container in the storage account in which the JDK is located."
+        },
+        {
+            "name": "commonVirtualPath",
+            "type": "string",
+            "label": "Common Virtual Path",
+            "required": true,
+            "visibleRule": "jdkSource == AzureStorage",
+            "helpMarkDown": "Path to the JDK inside the Azure storage container."
+        },
+        {
+            "name": "fileType",
```

Comment:
"jdkCompressedFileType". Do we need this or can we detect it and handle it in the task? In general lesser inputs we need the user to pass in, the better.